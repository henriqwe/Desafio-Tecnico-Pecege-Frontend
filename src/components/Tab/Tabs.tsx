import React, { memo, useEffect, useState } from 'react'
import * as S from './styles'

interface Props {
  children: JSX.Element | JSX.Element[]
  onTabSelected: (item: number) => void
}

export function Tab({ children, onTabSelected }: Props) {
  const [itemId, setItemId] = useState(0)

  useEffect(() => {
    onTabSelected(itemId)
  }, [itemId])

  return (
    <S.TabContainer>
      {React.Children.map(children, (child, index) => {
        return React.cloneElement(child, {
          onClick: () => {
            setItemId(index)
          },
          selected: itemId === index
        })
      })}
    </S.TabContainer>
  )
}

export const TabItem = memo(({ children, ...restProps }: any) => (
  <S.TabItem {...restProps}>{children}</S.TabItem>
))
