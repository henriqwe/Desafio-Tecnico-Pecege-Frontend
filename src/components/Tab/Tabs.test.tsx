import { fireEvent, render, waitFor } from '@testing-library/react'
import '@testing-library/jest-dom'
import { Tab, TabItem } from './Tabs'
import { TestProviders } from '&utils/tests/TestProviders'

const fnMoked = jest.fn()

describe('Tabs', () => {
  it('should render correctly snapshot', async () => {
    const { container, findByTestId } = render(
      <TestProviders>
        <Tab onTabSelected={fnMoked}>
          <TabItem data-testid={'tab-dados'}>Dados</TabItem>
          <TabItem data-testid={'tab-endereco'}>Endere√ßo</TabItem>
        </Tab>
      </TestProviders>
    )

    const tabDados = await findByTestId('tab-dados')

    await waitFor(async () => fireEvent.click(tabDados))

    expect(container).toMatchSnapshot()
    expect(fnMoked).toBeCalledWith(0)
  })
})
