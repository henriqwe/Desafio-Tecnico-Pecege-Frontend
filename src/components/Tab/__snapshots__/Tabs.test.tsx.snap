// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should render correctly snapshot 1`] = `
<div>
  <div
    class="sc-bcPKhP cTRxBW"
  >
    <div
      class="sc-grXZZQ giLQpt"
      data-testid="tab-dados"
    >
      Dados
    </div>
    <div
      class="sc-grXZZQ gbQVZO"
      data-testid="tab-endereco"
    >
      Endere√ßo
    </div>
  </div>
</div>
`;
